feat: enhanced undo ctrl panel

giuliaghisini opened the pull request:
Enhanced Undo controlpanel. 

Now it displays a loader while waiting for transactions from Plone. 
When you want to delete some transaction, a confirm popup asks the user to confirm undo action for selected items. 
Changed columns width to better display path.
Added select-all.
Added pagination. 
Added OP column.
Added some translations.  


https://github.com/user-attachments/assets/e4ee480a-0fa5-4b13-b0e7-aaa72963e0bd



Is it possibile to cherry-pick this PR in 17.x.x ?

stevepiercy commented:
For accessibility, the issue of standardizing colors came up in:

- https://github.com/plone/volto/pull/6544

The topic was added as an item to the PLIP.

- https://github.com/plone/volto/issues/6504

giuliaghisini commented:
> For accessibility, the issue of standardizing colors came up in:
> 
> * [Fix focus visibility for input components in site setup #6544](https://github.com/plone/volto/pull/6544)
> 
> The topic was added as an item to the PLIP.
> 
> * [Make CMS-UI inclusive and accessible #6504](https://github.com/plone/volto/issues/6504)

ok, ill'remove this fixes

stevepiercy commented on packages/volto/news/6545.feature:
```diff
@@ -0,0 +1 @@
+Improoved Undo control panel. @giuliaghisini
```
I pulled this from the PR description. I like explicit change logs.

```suggestion
Improved undo control panel. @giuliaghisini
- Now it displays a loader while waiting for transactions from Plone.
- When you want to delete some transactions, a confirmation popup asks the user to confirm the undo action for the selected items.
- Changed columns' width to better display path.
- Added some translations.
- Added style to accessibility focus on checkbox.
```

stevepiercy reviewed (changes requested):
Mostly comments on the `.pot` file to apply throughout the `.po` files, and expand the change log entry.

giuliaghisini commented:
> > ok, ill'remove this fixes
> 
> I think it's OK to leave them in, so we can see what you came up with and discuss it. I did not mean for you to delete them. Sorry for the miscommunication.

don't worry. I prefered to remove that styles because out of this pr. It's better to manage a11y styles in a single consistent pr.

tisto commented:
@giuliaghisini amazing work! Thank you, Giulia! 

This might be beyond the scope of this enhancement. However, I think we need to broaden the width of the control panels in Volto. The undo control panel, the redirection control panel, and other control panels that display lots of data in a table just do not have sufficient space. cc @sneridagh

giuliaghisini commented:
> This might be beyond the scope of this enhancement. However, I think we need to broaden the width of the control panels in Volto. The undo control panel, the redirection control panel, and other control panels that display lots of data in a table just do not have sufficient space. cc

i agree... had some difficults to gain width for path column.. 
maybe... for all controlpanels.. we could think to enlarge 'contents' width to 80% of monitor width for example..

giuliaghisini commented:
> @giuliaghisini amazing work! Thank you, Giulia!
> 
> This might be beyond the scope of this enhancement. However, I think we need to broaden the width of the control panels in Volto. The undo control panel, the redirection control panel, and other control panels that display lots of data in a table just do not have sufficient space. cc @sneridagh

tisto commented:
@giuliaghisini @sneridagh I would increase the size of the main area to 90% at least if that's technically possible. The control panels are for power users and they need as much space as possible.

giuliaghisini commented:
> @giuliaghisini @sneridagh I would increase the size of the main area to 90% at least if that's technically possible. The control panels are for power users and they need as much space as possible.

i've updated this pr where i have increased the width of controlpanels.
I've updated the description and the example video of this PR if you want to check the result

tisto commented:
@giuliaghisini looks great! Can wait to see this PR merged and released!

stevepiercy commented:
Would it be possible to deselect the checkbox for all items when paginating?

I also think I'm missing something. I can't see actual changes to content. I think it would be useful to preview a [diff of revisions like in Wordpress](https://wordpress.com/support/page-post-revisions/) before undoing a transaction, but that might not be possible or outside the scope of this PR.

ichim-david commented on packages/volto/news/6545.feature:
```diff
@@ -0,0 +1,6 @@
+- When you want to delete some transactions, a confirmation popup asks the user to confirm the undo action for the selected items.
+- Changed columns' width to better display path.
+- Added some translations.
+- Added style to accessibility focus on checkbox.
```
@giuliaghisini this entry is no longer valid after this revert commit of yours
https://github.com/plone/volto/pull/6545/commits/650f5b4cf4692e908fb396f359a392a6c492fc6c#diff-3a5f2698b3ccbd44cd87e2f818c58afbd7d9981a3c90afbaf56a80cf2c93f9c2

giuliaghisini replied:
yes, i've updated the description

giuliaghisini commented:
> Would it be possible to deselect the checkbox for all items when paginating?
> 
> I also think I'm missing something. I can't see actual changes to content. I think it would be useful to preview a [diff of revisions like in Wordpress](https://wordpress.com/support/page-post-revisions/) before undoing a transaction, but that might not be possible or outside the scope of this PR.

yes, i think this is outside the scope of this pr, maybe you could open an issue

sneridagh reviewed (approved):
LGTM, since it's CMSUI, it's not breaking, we can move forward.

@giuliaghisini some checks need attention.

stevepiercy commented:
Would it be possible to deselect the checkbox for all items when paginating?

giuliaghisini commented:
> Would it be possible to deselect the checkbox for all items when paginating?

this is an intentional feature, sometimes you want to undo a lot of actions in one single transaction..

giuliaghisini commented:
@sneridagh i dont'know what have to do to fix this check, i have this problem on another pr too : https://github.com/plone/volto/actions/runs/12686287603/job/35358301758?pr=6545

stevepiercy commented:
> > Would it be possible to deselect the checkbox for all items when paginating?
> 
> this is an intentional feature, sometimes you want to undo a lot of actions in one single transaction..

Let me try to explain it differently.

In the video, I see you click the top checkbox on Page 1 to select all items on the page. Then I see you go to the next page. I see you deselect the top checkbox, then reselect it, to select all items on the second page. I think that is one click too many and it looks a tiny bit weird to me.

Instead on the second page, I would like to see the top checkbox become deselected automatically. Then you can click it one time to select all items.

davisagli commented:
The UI with checkboxes is not ideal here. (I know it's the same that we always had in the ZMI; it was bad there too.) Usually it's not possible to undo an old transaction unless you're also undoing all the transactions that happened since then. So I think it would make more sense to use radio buttons, and undo all transactions between now and the one that the user selects.

giuliaghisini commented:
> The UI with checkboxes is not ideal here. (I know it's the same that we always had in the ZMI; it was bad there too.) Usually it's not possible to undo an old transaction unless you're also undoing all the transactions that happened since then. So I think it would make more sense to use radio buttons, and undo all transactions between now and the one that the user selects.

I've updated the PR and the behavior of checkboxes: now if you select a transaction, automatically will be checked all subsequents transactions until now as you suggested. 

See new bhv in this video:

https://github.com/user-attachments/assets/b1fd4d3e-efed-4dba-aafc-799630e936d4

stevepiercy commented:
This update makes a lot more sense to me. Thank you!
